thrust/thrust/system/cuda/memory_resource.h

```diff
@@ -88,8 +88,7 @@ inline cudaError_t CUDARTAPI cudaMallocManaged(void** ptr, std::size_t bytes)
 using device_memory_resource = detail::cuda_memory_resource<cudaMalloc, cudaFree, thrust::cuda::pointer<void>>;
 using managed_memory_resource =
   detail::cuda_memory_resource<detail::cudaMallocManaged, cudaFree, thrust::cuda::universal_pointer<void>>;
-using pinned_memory_resource =
-  detail::cuda_memory_resource<cudaMallocHost, cudaFreeHost, thrust::cuda::universal_pointer<void>>;
+using pinned_memory_resource = detail::cuda_memory_resource<cudaMallocHost, cudaFreeHost, void*>;
```

Comment:
The reason this let's `thrust::universal_host_pinned_vector` omit a kernel is that `thrust::detail::allocator_system<thrust::universal_host_pinned_vector<T>::allocator>` now reports the host and not the device system. Previously, because `thrust::cuda::universal_pointer` is an alias to `thrust::pointer<T, thrust::cuda_cub::tag, ...>`, the device system was used when operating on a pinned universal vector.

Reply:
I went with:
```c++
thrust::pointer<T, thrust::system::cpp::tag, ::cuda::std::add_lvalue_reference_t<T>>
```
and added a new alias `universal_host_pinned_pointer` to all backends.